### ISSUE TITLE
php 8.1, first pay, fails

### ISSUE THREAD

**Issue by adixon** [iATSPayments/com.iatspayments.civicrm#409]
Fails with a very unhelpful error message. Confirmed that the error does not occur when I switch the site back to php7.4

Using Drupal 9.5.3, CiviCRM 5.57.3

Also confirmed that legacy/original iATSpayments processor works on php8.1, so it's something specific to the 1st Pay thingy.

**Comment by KarinG** [comment]
Adding my notes: I have seen intermittent failures in Webform CiviCRM Github Actions re: 1st Pay (on PHP7.4) and in real life as well. I did contact iATS dev support - but did not get a response. I've now disabled the 1st Pay test in Webform CiviCRM Github Actions.

**Comment by adixon** [comment]
I think this might have been a misrepresentation. In the process of tracking this down, I did find that the latest iats was broken for firstpay, but it turned out to be this PR https://github.com/iATSPayments/com.iatspayments.civicrm/pull/350 which was fixing some currency bits, but was probably overzealous.

i.e. it wasn't php8 related.

I've just fixed that issue with this: https://github.com/iATSPayments/com.iatspayments.civicrm/pull/416

And now my firstpay test is working.
